### PR TITLE
Cleanup unused and duplicated code

### PR DESCRIPTION
**Pull request by tgrospic**
## Overview

As part of Block Receiver PR some additional cleanup is done which is now extracted in this PR without changes in any functionality.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/ValidatorIdentity.scala [hunk 1]
```diff
@@ -59,11 +56,6 @@ object ValidatorIdentity {
     )
   }
 
-  def getEnvVariablePassword[F[_]: Sync: EnvVars]: F[String] =
-    EnvVars[F].get(RNodeValidatorPasswordEnvVar) >>= (
-      _.liftTo(new Exception(s"Environment variable $RNodeValidatorPasswordEnvVar is unspecified"))
-    )
```
Comment (tgrospic) [inline casper/src/main/scala/coop/rchain/casper/ValidatorIdentity.scala:65]: This is not used now, but later we can provide more comprehensive way to manage private keys for validators.

**Review by stanislavlyalin — APPROVED** [review]
:+1:

**Review by nzpr — APPROVED** [review]
LGTM

**Comment by tgrospic** [comment]
bors merge

**Pull request merged**